### PR TITLE
Fix diff function

### PR DESCRIPTION
**Pull request by mdreizin**
Current `model.$diff()` doesn't work. This PR fixes that.

### CODE REVIEW

**Comment by mdreizin** [comment]
It works fine on dec env when using `grunt build && grunt karma:ci`.

**Comment by mdreizin** [comment]
@amcdnl @juristr Have you got a chance to look at this PR?

**Comment by juristr** [comment]
@mdreizin Sorry, not yet. We just set up [semantic-release](https://github.com/semantic-release/semantic-release) today. Have to verify it's correctly publishing to bower and npm before merging in this. 
I'll then use your PR to test it out. //cc @amcdnl

**Comment by mdreizin** [comment]
@juristr Thanks. I have never touched `semantic-release` and it looks great. Let's take a look how it will work :)

**Comment by juristr** [comment]
@mdreizin merged in master. semantic-release correctly uploaded `v1.0.1` to npm. 

I'm still working on automating release to bower which is kinda more complicated as it involves pushing the `dist` files into the repo back again. I don't want to have them in the master repo any more as it is a source of error during the release.

I'll keep u posted. Thx again for the PR. Even with tests :heart_eyes: love it :smile:

**Comment by juristr** [comment]
@mdreizin ok, publishing to bower fixed as well.

@amcdnl Woot! We now have fully automated deploys to npm and bower :heart_eyes: 

to know..

- removed `bower_components` folder and excluded it from git as it should be fetched ad-hoc at build-time
- configured travis and package.json to have semantic-release work properly
- added `.npmignore` to only include the required files in the npm package
- added [publish-latest](https://github.com/kentcdodds/publish-latest) package which is required to have Bower work.

I also added a more elaborate version of our [contribution guidelines](https://github.com/Swimlane/angular-model-factory/blob/master/CONTRIBUTING.md). The important part there is **that we need to use conventional commits** from now on so that semantic-release is able to automatically create changelogs.

So basically now when we merge in a branch/PR into master and push it up, Travis will start and build, exec the tests. Then semantic-release starts and does it work, determining the new version and publishing to npm.
Now, the problem was bower as it requires to have the `dist` files included in the repo. Apparently Kent C Dodds had the same problem a couple of days ago and created publish-release. What it does is basically to commit the `dist` + package.json (or whatever you specify) to a branch `latest` which it pushes back to the repo and tags it s.t. it can be retrieved by bower. Really sweet.

Guess I'll write the setup down in a blog post, for self-reference and 'cause it could be interesting to others as well. :smile:
